src/otx/backend/native/models/classification/factory.py

```diff
@@ -0,0 +1,289 @@
+    MobileNetV3MultilabelCls,
+    TimmModelMultilabelCls,
+    TVModelMultilabelCls,
+    VisionTransformerMultilabelCls,
+)
+
+if TYPE_CHECKING:
+    from lightning.pytorch.cli import LRSchedulerCallable, OptimizerCallable
+
+    from otx.backend.native.models.base import DataInputParams
+    from otx.backend.native.schedulers import LRSchedulerListCallable
+    from otx.metrics import MetricCallable
+    from otx.types.label import LabelInfoTypes
+
+
+class MobileNetV3:
+    """Factory class for MobileNetV3 models.
+
+    Args:
+        label_info (LabelInfoTypes): The label information.
+        data_input_params (DataInputParams): The data input parameters such as input size and normalization.
+        freeze_backbone (bool, optional): Whether to freeze the backbone during training. Defaults to False.
+            Note: only multiclass classification supports this argument.
+        model_name (str, optional): The model name. Defaults to "mobilenetv3_large".
+        task (Literal["multi_class", "multi_label", "h_label"], optional): The task type.
+            Can be "multi_class", "multi_label", or "h_label". Defaults to "multi_class".
+        optimizer (OptimizerCallable, optional): The optimizer callable. Defaults to DefaultOptimizerCallable.
+        scheduler (LRSchedulerCallable | LRSchedulerListCallable, optional): The learning rate scheduler callable.
+            Defaults to DefaultSchedulerCallable.
+        metric (MetricCallable, optional): The metric callable. Defaults to MultiClassClsMetricCallable.
+        torch_compile (bool, optional): Whether to compile the model using TorchScript. Defaults to False.
+    """
+
+    @overload
+    def __new__(
+        cls,
+        label_info: LabelInfoTypes,
+        data_input_params: DataInputParams,
+        task: Literal["multi_class", "multi_label", "h_label"] = "multi_class",
+        freeze_backbone: bool = False,
+        model_name: str = "mobilenetv3_large",
+        optimizer: OptimizerCallable = DefaultOptimizerCallable,
+        scheduler: LRSchedulerCallable | LRSchedulerListCallable = DefaultSchedulerCallable,
+        metric: MetricCallable = MultiClassClsMetricCallable,
+        torch_compile: bool = False,
+    ) -> MobileNetV3MulticlassCls | MobileNetV3MultilabelCls | MobileNetV3HLabelCls:
+        ...
+
+    def __new__(
+        cls,
+        task: Literal["multi_class", "multi_label", "h_label"] = "multi_class",
+        **kwargs,
+    ) -> MobileNetV3MulticlassCls | MobileNetV3MultilabelCls | MobileNetV3HLabelCls:
+        """Factory method to create MobileNetV3 models based on the task type."""
+        if task == "multi_class":
+            return MobileNetV3MulticlassCls(**kwargs)
+        if task == "multi_label":
+            return MobileNetV3MultilabelCls(**kwargs)
+        if task == "h_label":
+            return MobileNetV3HLabelCls(**kwargs)
+        msg = f"Unsupported task type: {task}"
+        raise ValueError(msg)
+
+
+class EfficientNet:
```

Comment:
There are two classes named `EfficientNet` in the codebase:

1. **Factory class** in `src/otx/backend/native/models/classification/factory.py` - This is the public API that creates task-specific EfficientNet models
2. **Backbone implementation** in `src/otx/backend/native/models/classification/backbones/efficientnet.py` - This is the internal neural network architecture

The factory class is exported as the main `EfficientNet` API, while the backbone class should be an internal implementation detail. I'm thinking renaming `src/otx/backend/native/models/classification/backbones/efficientnet.py::EfficientNet` to `EfficientNetFeatureExtractor` or `EfficientNetBackbone`?

Reply:
Done. For Mobilenet and VIT as well